src/module.c

```diff
@@ -3391,12 +3404,8 @@ int RM_StreamIteratorNextID(RedisModuleKey *key, RedisModuleStreamID *id, long *
         key->u.stream.currentid.ms = 0; /* for RM_StreamIteratorDelete() */
         key->u.stream.currentid.seq = 0;
         key->u.stream.numfieldsleft = 0; /* for RM_StreamIteratorNextField() */
-        if (id) {
-            id->ms = 0;
-            id->seq = 0;
-        }
-        if (numfields) *numfields = 0;
```

Comment:
i think it's a good idea to also zero the output arguments, not just rely on the return value.

Reply:
OK, I'll put this back again, tomorrow. (For reference: First I added this but then I removed it again because I thought it would be unexpected to touch the output args when ERR is returned.)